examples/Superbible/Vertex Colouring/src/main_loop.adb

```diff
@@ -0,0 +1,93 @@
+
+with Ada.Exceptions; use Ada.Exceptions;
+with Ada.Numerics.Generic_Elementary_Functions;
+with Ada.Text_IO; use Ada.Text_IO;
+
+with GL.Attributes;
+with GL.Buffers;
+with GL.Culling;
+with GL.Errors;
+with GL.Objects.Programs;
+with GL.Objects.Shaders;
+with GL.Objects.Vertex_Arrays;
+with GL.Toggles;
+with GL.Types; use GL.Types;
+with GL.Types.Colors;
+
+with Glfw.Input;
+with Glfw.Input.Keys;
+with Glfw.Windows;
+with Glfw.Windows.Context;
+
+with Program_Loader;
+with Utilities;
+
+procedure Main_Loop (Main_Window : in out Glfw.Windows.Window) is
+
+    package Math_Functions is new Ada.Numerics.Generic_Elementary_Functions (Single);
+    subtype tVec4f is GL.Types.Singles.Vector4;
+
+    Rendering_Program : GL.Objects.Programs.Program;
+    Vertex_Array      : GL.Objects.Vertex_Arrays.Vertex_Array_Object;
+
+    --  ------------------------------------------------------------------------
+
+    procedure Render_Triangle (Current_Time : Glfw.Seconds) is
+        use Math_Functions;
+        Back_Colour : GL.Types.Colors.Color := (0.5 * (1.0 + Sin (Single (Current_Time))),
```

Comment:
The code would be more readable if you calculated `Single (Current_Time)` only once and stored it in a variable.